amida-tech/recruitment-registry
# Feature/rr 906 participant dashboard stats

If this PR fixes a bug, you _must_ add test cases representative of the bug.

Please refer to [Tettra](https://app.tettra.co/teams/amida/pages/amida-pull-request-and-code-review-guide) for PR review guidelines.

#### What does this PR do?
- Adds a new API endpoint `/demographics` that outputs a list of all Participant user demographic data in JSON struct (example):
```
[
    {
        "Zip Code": "12345",
        "registrationDate": "2017-11-29",
        "Year of Birth": "1977"
    },
    {
        "Zip Code": "99987",
        "registrationDate": "2018-01-11",
        "Year of Birth": "1975"
    },
    {
        "Zip Code": "99998",
        "registrationDate": "2018-01-15",
        "Year of Birth": "1980"
    }
]
```

#### Related JIRA tickets:
- https://jira.amida-tech.com/browse/RR-906

#### How should this be manually tested?
- Create several Participant users and ensure that the associated `profile-survey` questions are answered (`zip` and `yearOfBirth`)
- As `admin`, GET `/demographics` to see the listed output

#### Background/Context
- [UPDATE 1/18/2017, I've made this scale beyond GAP ... however there is currently no support for `choice` or `choices` type questions]: NOTE that this implementation will not scale to clients beyond GAP due to their usage of the `profile-survey` functionality which these demographics are dependent on. There will need to be discussions on scaling out this functionality such as perhaps implementing a RecReg-side `demographics` table, or flattening the demographic data to a common key (i.e. `questionText`) while adding a `type` property for approximating intended data semantic.

#### Screenshots (if appropriate):
